D1371R0.pandoc

```diff
@@ -1116,15 +1116,26 @@ Each of the matchers and extractors can be found in the [Examples] section.
 the language. The intent of this proposal is to consolidate with the results
 of the decision of [@P1110R0].
 
-## Identifier and Expression Patterns Disambiguation
+## Pattern syntax and Expression Pattern Disambiguation
 
-Both identifier and expression patterns can contain an unqualified name. It can
-resolve into local variable name in case of [expression pattern](#expression-pat).
-And introduce new identifier into scope in case of
-[identifier pattern](#identifier-pat).
+It is clear that we want to be able to use arbitrary expressions in various
```

Comment:
I think it's worthwhile elaborating on this. Rust and Scala for example, do not allow arbitrary expressions as patterns. I'm rethinking whether it's actually crucial for us to support this.. 😕

Reply:
I can see how ability to call functions and make simple computations inside pattern can be deisrable. For the same reason we've put init-statement into if, for and etc. To be able to limit scope of conditions and condition variables.

I'll try to elaborate, but if we want to go the no-expression route - we may need to refocus this section later.